compiler/qsc_eval/src/lib.rs

```diff
@@ -39,67 +41,97 @@ use val::GlobalId;
 pub enum Error {
     #[error("array too large")]
     #[diagnostic(code("Qsc.Eval.ArrayTooLarge"))]
-    ArrayTooLarge(#[label("this array has too many items")] Span),
+    ArrayTooLarge(#[label("this array has too many items")] PackageSpan),
 
     #[error("invalid array length: {0}")]
     #[diagnostic(code("Qsc.Eval.InvalidArrayLength"))]
-    InvalidArrayLength(i64, #[label("cannot be used as a length")] Span),
+    InvalidArrayLength(i64, #[label("cannot be used as a length")] PackageSpan),
 
     #[error("division by zero")]
     #[diagnostic(code("Qsc.Eval.DivZero"))]
-    DivZero(#[label("cannot divide by zero")] Span),
+    DivZero(#[label("cannot divide by zero")] PackageSpan),
 
     #[error("empty range")]
     #[diagnostic(code("Qsc.Eval.EmptyRange"))]
-    EmptyRange(#[label("the range cannot be empty")] Span),
+    EmptyRange(#[label("the range cannot be empty")] PackageSpan),
 
     #[error("value cannot be used as an index: {0}")]
     #[diagnostic(code("Qsc.Eval.InvalidIndex"))]
-    InvalidIndex(i64, #[label("invalid index")] Span),
+    InvalidIndex(i64, #[label("invalid index")] PackageSpan),
 
     #[error("integer too large for operation")]
     #[diagnostic(code("Qsc.Eval.IntTooLarge"))]
-    IntTooLarge(i64, #[label("this value is too large")] Span),
+    IntTooLarge(i64, #[label("this value is too large")] PackageSpan),
 
     #[error("missing specialization: {0}")]
     #[diagnostic(code("Qsc.Eval.MissingSpec"))]
-    MissingSpec(String, #[label("callable has no {0} specialization")] Span),
+    MissingSpec(
+        String,
+        #[label("callable has no {0} specialization")] PackageSpan,
+    ),
 
     #[error("index out of range: {0}")]
     #[diagnostic(code("Qsc.Eval.IndexOutOfRange"))]
-    IndexOutOfRange(i64, #[label("out of range")] Span),
+    IndexOutOfRange(i64, #[label("out of range")] PackageSpan),
 
     #[error("negative integers cannot be used here: {0}")]
     #[diagnostic(code("Qsc.Eval.InvalidNegativeInt"))]
-    InvalidNegativeInt(i64, #[label("invalid negative integer")] Span),
+    InvalidNegativeInt(i64, #[label("invalid negative integer")] PackageSpan),
 
     #[error("output failure")]
     #[diagnostic(code("Qsc.Eval.OutputFail"))]
-    OutputFail(#[label("failed to generate output")] Span),
+    OutputFail(#[label("failed to generate output")] PackageSpan),
 
     #[error("qubits in gate invocation are not unique")]
     #[diagnostic(code("Qsc.Eval.QubitUniqueness"))]
-    QubitUniqueness(#[label] Span),
+    QubitUniqueness(#[label] PackageSpan),
 
     #[error("range with step size of zero")]
     #[diagnostic(code("Qsc.Eval.RangeStepZero"))]
-    RangeStepZero(#[label("invalid range")] Span),
+    RangeStepZero(#[label("invalid range")] PackageSpan),
 
     #[error("Qubit{0} released while not in |0⟩ state")]
     #[diagnostic(code("Qsc.Eval.ReleasedQubitNotZero"))]
-    ReleasedQubitNotZero(usize),
+    ReleasedQubitNotZero(usize, #[label("Qubit{0}")] PackageSpan),
```

Comment:
This is a new "feature": I added this span based on a suggestion from the bug bash. It'd be helpful to show the user _which_ qubit is being released.

Reply:
It used to show them, and it was removed before because it was just a the location in the stdlib (the id was still there). Now that we have nice stack traces, I agree it makes sense to put this back in. Thanks for doing that!